lib/src/view/broadcast/broadcast_tournament_screen.dart

```diff
@@ -47,58 +47,156 @@ class BroadcastTournamentScreen extends StatelessWidget {
+            child: ListTile(
+              dense: true,
+              title: DefaultTextStyle.merge(
+                style: Styles.sectionTitle,
+                child: Text(context.l10n.broadcastLiveBroadcasts),
               ),
             ),
           ),
-        );
-      },
-      error: (error, stackTrace) => Center(
-        child: Text(error.toString()),
+          SliverGrid.builder(
+            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+              crossAxisCount: 2,
+              crossAxisSpacing: 10,
+              mainAxisSpacing: 10,
+            ),
+            itemBuilder: (context, index) =>
+                BroadcastPicture(broadcast: broadcasts.value!.active[index]),
+            itemCount: broadcasts.value!.active.length,
+          ),
+          SliverToBoxAdapter(
+            child: ListTile(
+              dense: true,
+              title: DefaultTextStyle.merge(
+                style: Styles.sectionTitle,
+                child: const Text('Upcoming broadcasts'),
+              ),
+            ),
+          ),
+          SliverGrid.builder(
+            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+              crossAxisCount: 2,
+              crossAxisSpacing: 10,
+              mainAxisSpacing: 10,
+            ),
+            itemBuilder: (context, index) =>
+                BroadcastPicture(broadcast: broadcasts.value!.upcoming[index]),
+            itemCount: broadcasts.value!.upcoming.length,
+          ),
+          SliverToBoxAdapter(
+            child: ListTile(
+              dense: true,
+              title: DefaultTextStyle.merge(
+                style: Styles.sectionTitle,
+                child: const Text('Past broadcasts'),
+              ),
+            ),
+          ),
+          SliverGrid.builder(
+            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+              crossAxisCount: 2,
+              crossAxisSpacing: 10,
+              mainAxisSpacing: 10,
+            ),
+            itemBuilder: (context, index) => (broadcasts.isLoading &&
+                    index >= itemsCount - 10)
+                ? Shimmer(
+                    child: ShimmerLoading(
+                      isLoading: true,
+                      child: BroadcastPicture.loading(),
+                    ),
+                  )
+                : BroadcastPicture(broadcast: broadcasts.value!.past[index]),
+            itemCount: itemsCount,
+          ),
+          const SliverToBoxAdapter(
+            child: SizedBox(
+              height: 10,
+            ),
+          ),
+        ],
       ),
-      loading: () => const CircularProgressIndicator(),
     );
   }
 }
 
 class BroadcastPicture extends StatelessWidget {
   final Broadcast broadcast;
 
-  const BroadcastPicture({super.key, required this.broadcast});
+  const BroadcastPicture({required this.broadcast});
+
+  BroadcastPicture.loading() : broadcast = Broadcast.loading();
```

Comment:
It could use a const I think.

Reply:
I can't because the value I assign to broadcast is not a const value due to the presence of Datetime in `Broadcast`